### PR TITLE
Cleanup of inverted index

### PR DESCRIPTION
**Pull request by rschu1ze**
Various cosmetic and consistency cleanups for #38667.

Splitted into multiple smaller commits which should be simpler to double-check than the entire PR.

### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

### CODE REVIEW

File: src/Interpreters/GinFilter.h [hunk 1]
```diff
@@ -25,67 +29,54 @@ struct GinSegmentWithRowIDRange
...
     }
 
-    /// Const getter of terms(generated from the query string)
-    const std::vector<String>& getTerms() const { return terms;}
-
-    /// Check if the given postings list cache has matched rows by using the filter
-    bool match(const PostingsCache& postings_cache) const;
+    /// Getter
+    const String & getQueryString() const { return query_string; }
+    const std::vector<String> & getTerms() const { return terms;}
```
Comment (serxa) [inline src/Interpreters/GinFilter.h:73]:
```suggestion
    const std::vector<String> & getTerms() const { return terms; }
```

File: src/Storages/MergeTree/MergeTreeIndexInverted.cpp [hunk 2]
```diff
@@ -695,55 +690,55 @@ bool MergeTreeConditionGinFilter::tryPrepareSetGinFilter(
...
     return std::find(std::cbegin(index.column_names), std::cend(index.column_names), node->getColumnName()) != std::cend(index.column_names);
 }
 
-MergeTreeIndexPtr ginIndexCreator(
+MergeTreeIndexPtr invertedIndexCreator(
     const IndexDescription & index)
 {
     size_t n = index.arguments.empty() ? 0 : index.arguments[0].get<size_t>();
-    Float64 density = index.arguments.size() < 2 ? 1.0f : index.arguments[1].get<Float64>();
+    Float64 density = index.arguments.size() < 2 ? 1.0l : index.arguments[1].get<Float64>();
```
Comment (serxa) [inline src/Storages/MergeTree/MergeTreeIndexInverted.cpp:725]: We need literal of `double` type here? (`l` is for `long double`)
```suggestion
    Float64 density = index.arguments.size() < 2 ? 1.0 : index.arguments[1].get<Float64>();
```

File: src/Storages/MergeTree/MergeTreeIndexInverted.cpp [hunk 3]
```diff
@@ -776,10 +768,9 @@ void ginIndexValidator(const IndexDescription & index, bool /*attach*/)
     if (index.arguments.size() == 2 && (index.arguments[1].getType() != Field::Types::Float64 || index.arguments[1].get<Float64>() <= 0 || index.arguments[1].get<Float64>() > 1))
         throw Exception("The second Inverted index argument must be a float between 0 and 1.", ErrorCodes::INCORRECT_QUERY);
 
-    size_t ngrams = index.arguments.empty() ? 0 : index.arguments[0].get<size_t>();
-    Float64 density = index.arguments.size() < 2 ? 1.0f : index.arguments[1].get<Float64>();
-
     /// Just validate
+    size_t ngrams = index.arguments.empty() ? 0 : index.arguments[0].get<size_t>();
+    Float64 density = index.arguments.size() < 2 ? 1.0l : index.arguments[1].get<Float64>();
```
Comment (serxa) [inline src/Storages/MergeTree/MergeTreeIndexInverted.cpp:773]:
```suggestion
    Float64 density = index.arguments.size() < 2 ? 1.0 : index.arguments[1].get<Float64>();
```

**Review by serxa — APPROVED** [review]

File: src/Storages/MergeTree/MergeTreeIndexInverted.cpp (hunk 2 above)
Comment (rschu1ze) [reply src/Storages/MergeTree/MergeTreeIndexInverted.cpp:725]: You are right, `long double` is likely an 80bit float (extended precision) or even 128bit...

**Comment by rschu1ze** [comment]
ClickHouse Integration Tests (asan) [6/6]:
ClickHouse Integration Tests (release) [2/4]:
ClickHouse Integration Tests (tsan) [6/6]:
- unrelated, failures are also in master

ClickHouse Stress Test (ubsan): also unrelated

**Pull request merged**